File: SortingSearching_Java_RiyaBansal/L1/Bubble_Sort.java
package Sorting_Searching_1;

public class Bubble_Sort {

	public static void main(String[] args) {

		int[] arr = {5, 1, 4, 2, 8 };
		bubbleSort(arr);

//		for (int val : arr) {
//
//			System.out.print(val + " ");
//		}

	}
	
	//Time Complexity
	//O(n2) - Worst Case
	//O(n) - Best Case
	//Space Complexity
	//O(1)(In-Place Sorting Algo)
	private static void bubbleSort(int[] arr) {

		int n = arr.length;

		boolean swapped;
		for (int i = 0; i < n - 1; i++) {
			
			swapped = false;
			for (int j = 0; j < n - i - 1; j++) {

				
				if (arr[j] > arr[j + 1]) {

					int temp = arr[j];
					arr[j] = arr[j + 1];
					arr[j + 1] = temp;
					swapped = true;
				}
			}
			
			for (int k = 0; k < arr.length; k++) {

				System.out.print(arr[k] + " ");
			}
			System.out.println();
			
			if(swapped == false) {
				break;
			}
			
		}

	}

}

File: Pulkit_Chhabra/Sorting_And_Searching_Dec_2020/L1/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    for itr in range(n):
        swapped = False
        for i in range(n - itr - 1):
            if arr[i] > arr[i + 1]:
                temp = arr[i]
                arr[i] = arr[i + 1]
                arr[i + 1] = temp
                swapped = True

        if not swapped:
            break


def main():
    n = int(input())
    arr = list(map(int, input().split(' ')))

    bubble_sort(arr)

    for num in arr:
        print(num, end=" ")


if __name__ == "__main__":
    main()
